src/BinSkim.Rules/PERules/BA2006.BuildWithSecureTools.cs

```diff
@@ -66,6 +66,29 @@ public IEnumerable<IOption> GetOptions()
             new PerLanguageOption<AdvancedMitigations>(
                 AnalyzerName, nameof(AdvancedMitigationsEnforced), defaultValue: () => AdvancedMitigations.None);
 
+        public override void Initialize(BinaryAnalyzerContext context)
+        {
+            if (context.Policy == null) { return; }
+
+            StringToVersionMap currentMinimumToolVersions = context.Policy.GetProperty(MinimumToolVersions, cacheDefault: false);
+            StringToVersionMap defaultMinimumToolVersions = BuildMinimumToolVersionsMap();
+
+            // If user only overwrite part of the settings, currentMinimumToolVersions will only have those settings.
+            // This is to make sure other default settings are applied as well.
+            foreach (KeyValuePair<string, Version> defaultMinimumToolVersion in defaultMinimumToolVersions)
+            {
+                currentMinimumToolVersions.TryAdd(defaultMinimumToolVersion.Key, defaultMinimumToolVersion.Value);
+            }
+
+            context.Policy.SetProperty(
+                MinimumToolVersions,
+                currentMinimumToolVersions,
+                cacheDescription: false,
```

Comment:
I actually think cache description == true is correct here, but we can scrub for this later. btw - someone should go look at all our other rules for instances of this bug.